compiler/rustc_lint/src/builtin.rs

```diff
@@ -665,9 +665,7 @@ declare_lint_pass!(MissingCopyImplementations => [MISSING_COPY_IMPLEMENTATIONS])
 
 impl<'tcx> LateLintPass<'tcx> for MissingCopyImplementations {
     fn check_item(&mut self, cx: &LateContext<'_>, item: &hir::Item<'_>) {
-        if !(cx.effective_visibilities.is_reachable(item.owner_id.def_id)
-            && cx.tcx.local_visibility(item.owner_id.def_id).is_public())
-        {
+        if !cx.effective_visibilities.is_reachable(item.owner_id.def_id) {
```

Comment:
Is the local visibility check just redundant because effective visibility is always stricter?

Reply:
Yes, with 95a24c6ed489de0fa6e87ef99390a744c8d46ddc reachable to other crates implies `pub` again, so we revert various "adaptation" changes from https://github.com/rust-lang/rust/pull/111801 too.